cpp/src/traversal/sssp.cu

```diff
@@ -282,49 +282,30 @@ void SSSP<IndexType, DistType>::clean() {
  *
  * @file sssp.cu
  * --------------------------------------------------------------------------*/
-void sssp(Graph* gdf_G,
-                   gdf_column* sssp_distances,
-                   gdf_column* predecessors,
-                   const int source_vert) {
-
-  CUGRAPH_EXPECTS(gdf_G->adjList != nullptr, "Invalid API parameter");
-
-  void *sssp_dist_ptr, *pred_ptr;
-  // NOTE: gdf_column struct doesn't have a default constructor. So we can get
-  // garbage values for member fields. Right now, it's the caller's
-  // responsibility to ensure that the fields are initialised if the gdf_column
-  // ptr is not null
-  sssp_dist_ptr = (sssp_distances && sssp_distances->size)
-      ? sssp_distances->data
-      : nullptr;
-  pred_ptr =
-      (predecessors && predecessors->size) ? predecessors->data : nullptr;
-
-  CUGRAPH_EXPECTS(sssp_dist_ptr || pred_ptr, "Invalid API parameter");
-
-  if (sssp_dist_ptr) {
-    CUGRAPH_EXPECTS(!sssp_distances->valid, "Column must be valid");
-    // Integral types are possible, but we don't want to deal with overflow
-    // conditions right now
-    CUGRAPH_EXPECTS(sssp_distances->dtype == GDF_FLOAT32 ||
-                    sssp_distances->dtype == GDF_FLOAT64,
-                "Invalid API parameter");
-  }
+template <typename VT, typename WT>
+void sssp(Graph* graph,                                            
+          WT *distances,                                                  
+          VT *predecessors,                                               
+          const VT source_vertex) {
 
-  CUGRAPH_EXPECTS(gdf_G->adjList->offsets->dtype == GDF_INT32,
-              "Unsupported data type");
-  CUGRAPH_EXPECTS(gdf_G->adjList->indices->dtype == GDF_INT32,
-              "Unsupported data type");
+  CUGRAPH_EXPECTS(graph->adjList != nullptr, "Invalid API parameter");
 
-  if (pred_ptr)
-    CUGRAPH_EXPECTS(predecessors->dtype == gdf_G->adjList->indices->dtype,
-                "Unsupported data type");
+  CUGRAPH_EXPECTS(distances || predecessors, "Invalid API parameter, both outputs are nullptr");
 
-  if (sssp_dist_ptr)
-    CUGRAPH_EXPECTS(gdf_G->adjList->offsets->size - 1 <= sssp_distances->size,
-                "Invalid API parameter");
+  if (typeid(WT) != typeid(float) && typeid(WT) != typeid(double))
+    CUGRAPH_FAIL("Unsupported weight data type, please use float or double");
+  if (typeid(VT) != typeid(int) )
+    CUGRAPH_FAIL("Unsupported personalization_subset data type, please use int");
```

Comment:
should not refer to personalization_subset in error message

Reply:
fixed